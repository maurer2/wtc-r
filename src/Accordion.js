import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import styles from'./Accordion.module.css';

const Accordion = (props) => {
  const { accordionData, activeAccordionEntry, handleUpdateActiveAccordionEntry } = props;

  return (
    <dl className={ styles.accordion }>
      {
        accordionData.map((entry, index) => {
          const isActiveEntry = activeAccordionEntry === index;

          return (
            <Fragment key={ entry.id }>
              <dt
                className={ `${styles.title} ${isActiveEntry ? styles['title--is-active'] : ''} ` }
                onClick={ () => handleUpdateActiveAccordionEntry(index) }
              >
                { entry.question }
              </dt>
              <dd className={ `${styles.answer} ${isActiveEntry ? styles['answer--is-visible'] : ''} ` }>
                { entry.answer }
              </dd>
            </Fragment>
          )
        })
      }
    </dl>
  );
};

Accordion.propTypes = {
  accordionData: PropTypes.array,
  activeAccordionEntry: PropTypes.number,
  handleUpdateActiveAccordionEntry: PropTypes.func,
}

export default Accordion;