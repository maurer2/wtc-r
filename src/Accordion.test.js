import React from 'react';
import { shallow } from 'enzyme';
import Accordion from './Accordion';

describe('Accordion', () => {
  const accordionData = [
    {
      answer: 'Answer 1',
      id: '1',
      question: 'Question 1',
    },
    {
      answer: 'Answer 2',
      id: '2',
      question: 'Question 2',
    },
  ];
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Accordion accordionData={accordionData} />);
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();

    const firstTitle = wrapper.find('.title-link').first();

    firstTitle.simulate('click', { preventDefault: () => {} });

    expect(wrapper).toMatchSnapshot();
  });

  test('has wrapper class', () => {
    expect(wrapper.find('.accordion').exists()).toBe(true);
  });

  test('has question and answers', () => {
    expect(wrapper.find('.title').exists()).toBe(true);
    expect(wrapper.find('.answer').exists()).toBe(true);
  });

  test('each question has an answer answers', () => {
    const lengthQuestions = wrapper.find('.title').length;
    const lengthAnswers = wrapper.find('.answer').length;

    expect(lengthQuestions).toBe(lengthAnswers);
  });

  test('clicking on a title activates entry', () => {
    let firstQuestion = wrapper.find('.title').first();
    const firstTitle = wrapper.find('.title-link').first();

    expect(firstTitle.exists()).toBe(true);
    expect(firstQuestion.hasClass('title--is-active')).toBe(false);

    firstTitle.simulate('click', { preventDefault: () => {} });

    wrapper.update();

    firstQuestion = wrapper.find('.title').first();

    expect(firstQuestion.hasClass('title--is-active')).toBe(true);
  });
});
