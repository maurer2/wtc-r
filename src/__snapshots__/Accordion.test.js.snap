// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render correctly 1`] = `
<dl
  className="accordion"
>
  <dt
    aria-expanded={false}
    className="title"
  >
    <a
      className="title-link"
      href="/"
      onClick={[Function]}
    >
      Question 1
    </a>
  </dt>
  <dd
    className="answer"
  >
    Answer 1
  </dd>
  <dt
    aria-expanded={false}
    className="title"
  >
    <a
      className="title-link"
      href="/"
      onClick={[Function]}
    >
      Question 2
    </a>
  </dt>
  <dd
    className="answer"
  >
    Answer 2
  </dd>
</dl>
`;

exports[`Accordion should render correctly 2`] = `
<dl
  className="accordion"
>
  <dt
    aria-expanded={true}
    className="title title--is-active"
  >
    <a
      className="title-link"
      href="/"
      onClick={[Function]}
    >
      Question 1
    </a>
  </dt>
  <dd
    className="answer answer--is-visible"
  >
    Answer 1
  </dd>
  <dt
    aria-expanded={false}
    className="title"
  >
    <a
      className="title-link"
      href="/"
      onClick={[Function]}
    >
      Question 2
    </a>
  </dt>
  <dd
    className="answer"
  >
    Answer 2
  </dd>
</dl>
`;
